### PR TITLE
Updates documentation on dynamic item classes.

### LINKED ISSUE #398: Dynamic Creation of Item Classes
I have a need to dynamically create item classes based on user input.  It's not clear to me how to approach that with Scrapy, it seems to assume that a developer will be coding item classes manually.

I created this utility function:

```
from scrapy.item import DictItem, Field

def create_item_class(class_name,field_list):
    field_dict = {}
    for field_name in field_list:
        field_dict[field_name] = Field()

    return type(class_name,DictItem,field_dict)
```

Not sure if this is the best way to approach this kind of problem with Scrapy, but I leave it here for your consideration.  Thanks!

### PR DESCRIPTION
**Pull request by barraponto**
Fixes #398
Used dict comprehension for more pythonic code.

### CODE REVIEW

File: docs/topics/practices.rst [hunk 1]
```diff
@@ -199,8 +199,6 @@ classes instead of manually coding them.
     from scrapy.item import DictItem, Field
 
     def create_item_class(class_name, field_list):
-        field_dict = {}
-        for field_name in field_list:
-            field_dict[field_name] = Field()
+        fields = {field_name: Field() for field_name in field_list}
 
-        return type(class_name, (DictItem,), field_dict)
+        return type(class_name, (DictItem,), {'fields': fields})
```
Comment (nramirezuy) [inline docs/topics/practices.rst]: `return type(class_name, (DictItem,), {'fields': fields})` this is wrong; should be `return type(class_name, (DictItem,), fields)`

Comment (barraponto) [inline docs/topics/practices.rst]: Oh, it is needed at least since 0.22.2 https://github.com/scrapy/scrapy/issues/398#issuecomment-40894736

Comment (nramirezuy) [inline docs/topics/practices.rst]: You are right, I didn't noticed the `type`.

**Comment by barraponto** [comment]
This should be backported to 0.24 too, since the docs are misleading there...

**Pull request merged**